src/vs/workbench/common/resources.ts

```diff
@@ -37,19 +39,22 @@ export class ResourceContextKey implements IContextKey<URI> {
 		this._filenameKey = ResourceContextKey.Filename.bindTo(contextKeyService);
 		this._langIdKey = ResourceContextKey.LangId.bindTo(contextKeyService);
 		this._resourceKey = ResourceContextKey.Resource.bindTo(contextKeyService);
+		this._extensionKey = ResourceContextKey.Extension.bindTo(contextKeyService);
 	}
 
 	set(value: URI) {
 		this._resourceKey.set(value);
 		this._schemeKey.set(value && value.scheme);
 		this._filenameKey.set(value && basename(value.fsPath));
 		this._langIdKey.set(value && this._modeService.getModeIdByFilenameOrFirstLine(value.fsPath));
+		this._extensionKey.set(value && paths.extname(value.fsPath));
```

Comment:
Way to go but we to think if folks expect `.do` from `foo.do` or `do`...

Reply:
I think typical behavior is to return extension with the dot. At least that's what most standard libraries I know do - Node's `path.extname`, .Net Framework `Path.GetExtension`, or Ruby's `File.extname`

Reply:
More interesting question is if users expects it to be normalized to lower case (i.e if "ABC.FSX" should return ".FSX" or just ".fsx")

Reply:
Good questions... I'd say we have it "as is". I can see how a "normalized" version can help but that applies to many values, e.g. filename